modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmControllerService.java

```diff
@@ -78,9 +85,53 @@
 	public static final String HTTP_PROTOCOL = "HTTP";
 	public static final String MQTT_PROTOCOL = "MQTT";
 
+	private static ConcurrentHashMap<String, String> deviceToIpMap = new ConcurrentHashMap<String, String>();
+	private static XMPPClient xmppClient;
+	private static MQTTClient mqttClient;
+	private static final String mqttServerSubscribeTopic = "wso2/iot/+/" + VirtualFireAlarmConstants.DEVICE_TYPE + "/+/reply";
+	private static final String iotServerSubscriber = "IoT-Server";
+
+	static{
+		String xmppServer = XmppConfig.getInstance().getXmppControlQueue().getServerURL();
+		int indexOfChar = xmppServer.lastIndexOf('/');
+		if (indexOfChar != -1) {
+			xmppServer = xmppServer.substring((indexOfChar + 1), xmppServer.length());
+		}
+
+		int xmppPort = Integer.parseInt(XmppConfig.getInstance().getSERVER_CONNECTION_PORT());
+		xmppClient = new XMPPClient(xmppServer, xmppPort) {
+			@Override
+			protected void processXMPPMessage(Message xmppMessage) {
+
```

Comment:
As we are not processing incoming XMPP message in here, there is no any method body. But we have to implement this method as it is defined in the interface class.